migtools/oadp-non-admin
# Allow Non-Admin user to create Backup Storage Locations - fixes #36

## Why the changes were made

Fix #36, also #57

## How to test the changes made

1. Create as usual OADP with NAC.
2. Create user secret in the user's namespace similarly to the S3 secret for DPA:

```shell
$ oc create secret generic nac-credentials -n <USERS_NAMESPACE> --from-file cloud=credentials-velero
```
3. Create Non Admin BSL pointing to the above credential in the user's namespace, example spec part:
```yaml

apiVersion: oadp.openshift.io/v1alpha1
kind: NonAdminBackupStorageLocation
metadata:
  name: example
  namespace: nactest
spec:
  config:
    region: eu-central-1
  credential:
    key: cloud
    name: nac-credentials
  objectStorage:
    bucket: my-bucket
    prefix: some-prefix
  provider: aws

```

4. Check the NAC logs, the NaBSL object (if it was correctly reconciled and updated with status from Velero BSL)

5...6...7... Do break things, such as improper Secret data, non-existing secret, removal of the secret from the OADP namespace or the user namespace, removal of the BSL from the OADP namespace and see if everything reconciles as expected.

Read [design](https://github.com/mpryc/oadp-non-admin/blob/nonadminstoragelocation_controller/docs/design/non_admin_backupstoragelocation.md)
